src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -235,15 +235,25 @@ protected Task Stop()
             string helpParams,
             RequestContext<object> requestContext)
         {
-            if (helpParams == null) { helpParams = "get-help"; }
-
-            var psCommand = new PSCommand();
-            psCommand.AddCommand("Get-Help");
-            psCommand.AddArgument(helpParams);
-            psCommand.AddParameter("Online");
-
-            await editorSession.PowerShellContext.ExecuteCommand<object>(psCommand);
-
+            if (helpParams == null) { helpParams = "Get-Help"; }
+            string script = @"
+[CmdletBinding()]
+param (
+    [String]$CommandName
+)
+Try {
+    Microsoft.PowerShell.Core\Get-Command $CommandName -ErrorAction Stop | Out-Null
+} catch [System.Management.Automation.CommandNotFoundException] {
+    $PSCmdlet.ThrowTerminatingError($PSItem)
```

Comment:
in other words you might be able to ditch the try/catch all together.

Reply:
I like it with the try catch because without it looks like this:
![image](https://user-images.githubusercontent.com/30301021/45072264-1beb0880-b08f-11e8-991b-fff2fc3e2abe.png)

and with it looks like this:
![image](https://user-images.githubusercontent.com/30301021/45072413-aaf82080-b08f-11e8-9fd0-e0950b2818a1.png)

But I will go with whichever you and @SeeminglyScience want.

Reply:
@tylerl0706 the benefit of using `ThrowTerminatingError` over letting the exception propagate is that it changes the "context" of the error from the callee(?) to the caller.

Here's a little function you can use to check it out

```powershell
function Test-TerminatingError {
    [CmdletBinding()]
    param(
        [Parameter(Position = 0, Mandatory)]
        [ValidateSet('NormalThrow', 'ThrowTerminatingError')]
        [string] $ThrowType
    )
    end {
        if ($ThrowType -eq 'NormalThrow') {                j
            Get-Command DefinitelyDoesntExist -ErrorAction Stop
        }

        try {
            Get-Command DefinitelyDoesntExist -ErrorAction Stop
        } catch {
            $PSCmdlet.ThrowTerminatingError($PSItem)
        }
    }
}
```

If `NormalThrow` is specified, this is what is displayed

```raw
Get-Command : The term 'DefinitelyDoesntExist' is not recognized as the name of a cmdlet, function, script file, or
operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try
again.
At line:11 char:13
+             Get-Command DefinitelyDoesntExist -ErrorAction Stop
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (DefinitelyDoesntExist:String) [Get-Command], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException,Microsoft.PowerShell.Commands.GetCommandCommand
```

It shows the line in the function/script itself that failed, which is implementation detail the user doesn't care about.  Contrast that to this:

```raw
Test-TerminatingError : The term 'DefinitelyDoesntExist' is not recognized as the name of a cmdlet, function, script
file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct
and try again.
At line:1 char:1
+ Test-TerminatingError ThrowTerminatingError
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (DefinitelyDoesntExist:String) [Test-TerminatingError], CommandNotFoundE
   xception
    + FullyQualifiedErrorId : CommandNotFoundException,Test-TerminatingError
```

It also lets you fully (well mostly) customize the displayed error by creating your own `ErrorRecord`

All that said, @corbob that is definitely not how I expected that to look...  something in our host does *not* handle that right...

We should still stick with it, but we should definitely look into why that's doing that.

Reply:
@SeeminglyScience should we be opening an issue for "something in our host does not handle that right...
...
we should definitely look into why that's doing that." ?